src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotheap/HighHeapUsageYoungGenRca.java

```diff
@@ -118,6 +139,111 @@ protected void remove(SlidingWindowData e) {
+                                                               double avgFullGCTime) {
+    ResourceContext context = new ResourceContext(State.UNHEALTHY);
+    HotResourceSummary summary = null;
+    boolean unhealthy = true;
+
+    // Check if the RCA is unhealthy
+    if (fullGcTimeTooHigh(avgFullGCTime)) {
+      summary = new HotResourceSummary(FULL_GC_PAUSE_TIME, getFollowerCheckTimeoutMs(), avgFullGCTime,
+          PROMOTION_RATE_SLIDING_WINDOW_IN_MINS * 60);
+      PerformanceAnalyzerApp.RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+          RcaVerticesMetrics.YOUNG_GEN_RCA_NAMED_COUNT, FULL_GC_TIME_TOO_HIGH, 1);
+    } else if (promotionRateTooHigh(avgPromotionRate, this.lowerBoundThreshold)) {
+      //check to see if the value is above lower bound thres
+      summary = new HotResourceSummary(YOUNG_GEN_PROMOTION_RATE,
+          promotionRateThreshold * this.lowerBoundThreshold, avgPromotionRate,
+          PROMOTION_RATE_SLIDING_WINDOW_IN_MINS * 60);
+      PerformanceAnalyzerApp.RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+          RcaVerticesMetrics.YOUNG_GEN_RCA_NAMED_COUNT, PROMOTION_RATE_TOO_HIGH, 1);
+    } else if (youngGcTimeTooHigh(avgYoungGCTime)) {
+      summary = new HotResourceSummary(MINOR_GC_PAUSE_TIME, youngGenGcTimeThreshold, avgYoungGCTime,
+          PROMOTION_RATE_SLIDING_WINDOW_IN_MINS * 60);
+      PerformanceAnalyzerApp.RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+          RcaVerticesMetrics.YOUNG_GEN_RCA_NAMED_COUNT, YOUNG_GC_TIME_TOO_HIGH, 1);
+    } else if (prematurePromotionTooHigh(avgGarbagePromoted)) {
+      summary = new HotResourceSummary(YOUNG_GEN_PROMOTION_RATE, garbagePromotionPctThreshold,
+          avgGarbagePromoted, PROMOTION_RATE_SLIDING_WINDOW_IN_MINS * 60);
+      PerformanceAnalyzerApp.RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+          RcaVerticesMetrics.YOUNG_GEN_RCA_NAMED_COUNT, PREMATURE_PROMOTION_TOO_HIGH, 1);
+    } else {
+      unhealthy = false;
+      context = new ResourceContext(State.HEALTHY);
+    }
+
+    if (unhealthy) {
+      LOG.debug("avgPromotionRate = {} , avgGCTime = {}, avgGarbagePromoted = {}, avgFullGcTime = {},",
+          avgPromotionRate, avgYoungGCTime, avgGarbagePromoted, avgFullGCTime);
+      PerformanceAnalyzerApp.RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+          RcaVerticesMetrics.NUM_YOUNG_GEN_RCA_TRIGGERED, "", 1);
+    }
+
+    return new ResourceFlowUnit<>(this.clock.millis(), context, summary);
+  }
+
+  /**
+   * computes the amount of data promoted and reclaimed by the garbage collector
+   * @param currOldGen the current occupancy of the old generation in bytes
+   * @param currTimeStamp the current timestamp in UNIX epoch milliseconds
+   */
+  private void computePromotionHealth(double currOldGen, long currTimeStamp) {
+    if (currOldGen > maxOldGen) {
+      maxOldGen = currOldGen;
+    }
+    double promoted = currOldGen - prevOldGen;
+    if (promoted >= 0) {
```

Comment:
I suggest that we should also check if full GC count > 0 in the last 5 sec time interval

Reply:
Checked in the else case in the upcoming PR